File: frontend/lib/documents.ts
const BASE = '/api/admin'

export type DocType =
  | 'cni' | 'passeport' | 'contrat' | 'garantie'
  | 'assurance' | 'fiscal' | 'medical' | 'autre'

export type ExpiryStatus = 'ok' | 'warning' | 'expired' | 'no_date'

export type Document = {
  id: number
  titre: string
  type: DocType
  notes: string
  date_expiration: string | null
  date_emission: string | null
  organisme: string
  fichier_url: string | null
  tags: string  // JSON
  statut_expiration: ExpiryStatus
  created_at: string
  updated_at: string
}

export const TYPE_LABELS: Record<DocType, string> = {
  cni: 'CNI',
  passeport: 'Passeport',
  contrat: 'Contrat',
  garantie: 'Garantie',
  assurance: 'Assurance',
  fiscal: 'Fiscal',
  medical: 'Médical',
  autre: 'Autre',
}

export const ALL_TYPES: DocType[] = ['cni', 'passeport', 'contrat', 'garantie', 'assurance', 'fiscal', 'medical', 'autre']

const json = (r: Response) => r.json()
const jsonHeaders = { 'Content-Type': 'application/json' }

export const fetchDocuments = (opts?: { type?: DocType; q?: string }): Promise<Document[]> => {
  const p = new URLSearchParams()
  if (opts?.type) p.set('type', opts.type)
  if (opts?.q) p.set('q', opts.q)
  const qs = p.toString()
  return fetch(`${BASE}/documents${qs ? '?' + qs : ''}`).then(json)
}

export const fetchEcheances = (days = 90): Promise<Document[]> =>
  fetch(`${BASE}/documents/echeances?days=${days}`).then(json)

export const createDocument = (data: Partial<Document> & { titre: string }): Promise<Document> =>
  fetch(`${BASE}/documents`, {
    method: 'POST',
    headers: jsonHeaders,
    body: JSON.stringify(data),
  }).then(json)

export const updateDocument = (id: number, patch: Partial<Document>): Promise<Document> =>
  fetch(`${BASE}/documents/${id}`, {
    method: 'PATCH',
    headers: jsonHeaders,
    body: JSON.stringify(patch),
  }).then(json)

export const deleteDocument = (id: number): Promise<void> =>
  fetch(`${BASE}/documents/${id}`, { method: 'DELETE' }).then(() => undefined)

File: backend/app/models/documents.py
"""Module Documents/Administratif (#548)."""
import datetime as dt
from app.core.timeutil import utcnow

from sqlmodel import Field, SQLModel


class Document(SQLModel, table=True):
    __tablename__ = "document"
    id: int | None = Field(default=None, primary_key=True)
    titre: str
    type: str = "autre"       # cnI | passeport | contrat | garantie | assurance | fiscal | medical | autre
    notes: str = ""
    date_expiration: dt.date | None = None
    date_emission: dt.date | None = None
    organisme: str = ""       # mairie, banque, assureur…
    fichier_url: str | None = None  # chemin local ou URL
    tags: str = "[]"          # JSON list
    created_at: dt.datetime = Field(default_factory=utcnow)
    updated_at: dt.datetime = Field(default_factory=utcnow)
